lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -542,6 +543,47 @@ def _delete_environment(self, environment: Environment, log: Logger) -> None:
             else:
                 log.debug("not wait deleting")
 
+    def _delete_boot_diagnostic_container(
+        self, resource_group_name: str, log: Logger
+    ) -> None:
+        compute_client = get_compute_client(self)
+        vms = compute_client.virtual_machines.list(resource_group_name)
+        for vm in vms:
+            diagnostic_data = (
+                compute_client.virtual_machines.retrieve_boot_diagnostics_data(
+                    resource_group_name=resource_group_name, vm_name=vm.name
+                )
+            )
+            if diagnostic_data:
+                # https://storageaccountname.blob.core.windows.net:443/bootdiagnostics-node0-30779088-9b10-4074-8c27-98b91f1d8b70/node-0.30779088-9b10-4074-8c27-98b91f1d8b70.serialconsole.log?sv=2018-03-28&sr=b&sig=mJEsvk9WunbKHfBs1lo1jcIBe4owq1brP8Kw3qXTQJA%3d&se=2021-09-14T08%3a55%3a38Z&sp=r # noqa: E501
+                blob_uri = diagnostic_data.console_screenshot_blob_uri
+                if blob_uri:
+                    pattern = re.compile(
```

Comment:
Move the pattern into class level, so it doesn't need to be compiled every time.

Reply:
fixed